src/components/grid/grid.ts

```diff
@@ -328,6 +329,10 @@ export class Grid extends Component<Props, SpreadsheetChildEnv> {
     return this.hoveredCell.col === col && this.hoveredCell.row === row;
   }
 
+  private getGridRect(): Rect {
```

Comment:
in master , we introduced  the getter`getVisibleRectWithoutHeaders`  which I think does what you are looking for. it might be better to backport that getter, we never know if we might need it for ulterior bugfixes

Reply:
If I got it right, we need to include headers in the calculation. The discussion is about where to show the context menu when pressing Shift + F10. I'm just trying to replicate the behavior that's already in the main version.

![Screenshot from 2024-04-04 11-17-52](https://github.com/odoo/o-spreadsheet/assets/121081869/f6095286-a350-4ee7-b221-83df86d5cfff)